plugins/hypervisors/xenserver/src/main/java/com/cloud/hypervisor/xenserver/resource/CitrixResourceBase.java

```diff
@@ -5611,4 +5612,122 @@ public boolean attachConfigDriveToMigratedVm(Connection conn, String vmName, Str
+            localDir = "/var/cloud_mount/" + UUID.nameUUIDFromBytes(secondaryStorageMountPath.getBytes());
+            mountNfs(conn, secondaryStorageMountPath, localDir);
+            String diagnosticsFileDir = "diagnostics_data";
+
+            String secStoragePath = localDir + "/" + diagnosticsFileDir;
+            Pair<Boolean, String> copyDetails = copyFilesToSecondaryStorage(vmIP, fileName, secStoragePath);
+            success = copyDetails.first();
+
+            if (!success) s_logger.error("failed to copy zip file to sec storage: " + copyDetails.second());
+
+        } catch (URISyntaxException e) {
+            throw new CloudRuntimeException("URI exception caught during decoding of secondary storage URI " + secondaryStorageUrl, e);
+        } finally {
+            if (localDir != null) umountNfs(conn, secondaryStorageMountPath, localDir);
+            // Cleanup file in system VM
+            cleanUpFileInVm(vmIP, fileName);
+        }
+        return new Answer(cmd, success, "");
+    }
+
+    /**
+     * Get Diagnostics Data API
+     * Copy zip file from system vm directly to diagnostics folder in secondary storage
+     * Secondary storage needs to be mounted on hypervisor host at this point
+     *
+     * @param vmIP           vm ssh IP address
+     * @param fileName       /root/diagnostics_files_xxxx.tar
+     * @param secStoragePath the path to 'diagnostics_data' directory in mounted secondary storage e.x /var/cloud_mout/uuid/diagnostics_data
+     *                       directory is created if it does not exist
+     * @return
+     */
+    protected Pair<Boolean, String> copyFilesToSecondaryStorage(String vmIP, String fileName, String secStoragePath) {
+        final String DEFAULT_PORT = "3922";
+        final String PERM_KEY = "/root/.ssh/id_rsa.cloud";
+
+        // scp -P 3922 -i /root/.ssh/id_rsa.cloud root@<vmIP>:<fileName> <sec_storage_path>/diagnostics_data
+        String cmdLine = String.format("mkdir -p %s && /usr/bin/scp -P %s -i %s root@%s:%s %s/.", secStoragePath, DEFAULT_PORT, PERM_KEY, vmIP, fileName, secStoragePath);
+        Pair<Boolean, String> executionResult = null;
+        try {
+            executionResult = xenServerUtilitiesHelper.executeSshWrapper(_host.getIp(), 22, _username, null, _password.peek(), cmdLine);
+        } catch (Exception e) {
+            String msg = String.format("Failed to copy %s to secondary storage due to exception!! ", fileName);
+            s_logger.error(msg);
+            throw new CloudRuntimeException(msg, e);
+        } finally {
+            cleanUpFileInVm(vmIP, fileName);
+        }
+        return executionResult;
+    }
+
+    /**
+     * Get Diagnostics Data API
+     * Delete zip file from vm
+     *
+     * @param vmIP     ssh IP
+     * @param fileName filepath inside system vm, /root/diagnostics_files_xxxx.tar
+     */
+    protected void cleanUpFileInVm(String vmIP, String fileName) {
+        final String PERM_KEY = "/root/.ssh/id_rsa.cloud";
```

Comment:
please remove this here

Reply:
done, extracted Zip file in system vm file cleanup to Network element command which makes it hypervisor agnostic